File: weyl-application/src/main/java/io/quartic/weyl/WeylConfiguration.java
package io.quartic.weyl;

import io.dropwizard.Configuration;
import io.quartic.catalogue.CatalogueClientConfiguration;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

public class WeylConfiguration extends Configuration {
    @Valid
    @NotNull
    private CatalogueClientConfiguration catalogue;

    @Valid
    @NotNull
    private String howlStorageUrl;

    public CatalogueClientConfiguration getCatalogue() {
        return this.catalogue;
    }

    public void setCatalogue(CatalogueClientConfiguration catalogue) {
        this.catalogue = catalogue;
    }

    public String getHowlStorageUrl() {
        return howlStorageUrl;
    }

    public void setHowlStorageUrl(String howlStorageUrl) {
        this.howlStorageUrl = howlStorageUrl;
    }
}

File: weyl-application/src/main/java/io/quartic/weyl/WeylConfiguration.kt
package io.quartic.weyl

import io.quartic.catalogue.CatalogueClientConfiguration
import io.quartic.catalogue.api.model.DatasetNamespace
import io.quartic.common.application.ConfigurationBase
import io.quartic.weyl.core.model.Tag

data class WeylConfiguration(
        val catalogue: CatalogueClientConfiguration,
        val howlStorageUrl: String,
        val rainWsUrlRoot: String,
        val importNamespaceRules: Map<DatasetNamespace, List<Tag>>,
        val exportNamespace: DatasetNamespace,
        val map: MapConfig
) : ConfigurationBase()

data class MapConfig(
        val lat: Double,
        val lng: Double,
        val zoom: Double    // Mapbox-specific (0-20)
)
